src/components/EditableHeading/EditableHeading.tsx

```diff
@@ -0,0 +1,168 @@
+import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
+import React, { forwardRef, useLayoutEffect, useMemo, useRef, useState } from "react";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import Heading from "../Heading/Heading";
+import { withStaticProps } from "../../types";
+import styles from "./EditableHeading.module.scss";
+import { HeadingType, HeadingWeight } from "../Heading/HeadingConstants";
+import cx from "classnames";
+import { getStyle } from "../../helpers/typesciptCssModulesHelper";
+import { camelCase } from "lodash-es";
+import HiddenInputPlaceholder from "./HiddenInputPlaceholder";
+import { keyCodes } from "../../constants";
+import useMergeRefs from "../../hooks/useMergeRefs";
+import useClickableProps from "../../hooks/useClickableProps/useClickableProps";
+
+export interface EditableHeadingProps extends VibeComponentProps {
+  value: string;
+  type?: HeadingType;
+  weight?: HeadingWeight;
+  editable?: boolean;
+  onChange?: (value: string) => void;
+  focusOnMount?: boolean;
+  selectOnMount?: boolean;
+  ariaLabel?: string;
+}
+
+const EditableHeading: React.FC<EditableHeadingProps> & {
+  types?: typeof Heading.types;
+  weights?: typeof Heading.weights;
+} = forwardRef(
+  (
+    {
+      id,
+      className,
+      "data-testid": dataTestId,
+      value,
+      type = Heading.types.H1,
+      weight = HeadingWeight.NORMAL,
+      editable = true,
+      onChange = () => {},
+      focusOnMount = true,
```

Comment:
should probably be `false`

Reply:
That's the default in the old component, I'll make sure with Meirav though

Reply:
It's more of a business-decision, so I'd suggest keeping the component non-opinionated and allow the consumer to decide if he wants to focus

Reply:
Talked to Meirav about this and she said that editing the text is the purpose so it should be focused by default, and also she suggested to keep it the same as the old component's default

Reply:
None of the html elements are focused by default. This is always an opt-in feature and I can think of product cases that doesn’t require editing by default (board header). So let’s avoid from making it on by default

Reply:
Just to be clear - the `onMount` refers to the editing mode, not the view mode. E.g. you can on the component, the input is shown, and only then you focus on the input

Reply:
So why do you need to props. If it’s in edit mode - isn’t it always focused?

Reply:
Which two props? I'm referring only to the focusOnMount prop, it can be disabled when you want to control the focus externally for example, or if you want to select the text on mount which is also a feature we provide